pkg/retry/retry.go

```diff
@@ -52,11 +52,20 @@ func (e *ErrorSet) Append(err error) error {
 	defer e.mu.Unlock()
 
 	if e.errs == nil {
-		e.errs = make(map[string]error)
+		e.errs = []error{}
 	}
 
-	if _, ok := e.errs[err.Error()]; !ok {
-		e.errs[err.Error()] = err
+	ok := false
+
+	for _, existingErr := range e.errs {
+		if err == existingErr {
```

Comment:
should we compare `err.Error() == existingErr.Error()` ?

as errors might different by pointer, but same by value? (if say, generated with `fmt.Errrof()`)